tests/integration_tests/conftest.py

```diff
@@ -119,49 +129,56 @@ def get_validated_source(
 
 
 @pytest.fixture(scope="session")
-def setup_image(session_cloud: IntegrationCloud, request):
-    """Setup the target environment with the correct version of cloud-init.
+def setup_image(session_cloud: IntegrationCloud):
```

Comment:
Just for some context, this used to be a fixture that allowed you to build a new snapshot in individual tests. That turned out to be a stupid and unnecessary footgun, so a while ago I made this a session fixture to set up the image for the entire suite. Usages of it in tests are currently no-ops. It really doesn't need to be a fixture at all anymore and can just be part of the session cloud setup. I'm not mentioning this to necessarily change anything here, but just for some context in case that makes thinking through and/or refactoring these changes easier. The separation that exists now between these two fixtures makes ownership between the two a bit jumbled.

Reply:
Thanks, I was wondering if that was how this developed over time.